src/hooks/useFavorites.ts

```diff
@@ -0,0 +1,34 @@
+import { ImagePerBreed } from "../api/images/types";
+import { FAVORITES_CONST } from "../config/consts";
+import useStorageArray from "./useStorageArray";
+
+const useFavorites = () => {
+  const {
+    items: favorites,
+    addItem,
+    removeItem,
+  } = useStorageArray(FAVORITES_CONST);
+
+  const isFavorite = (id: ImagePerBreed["id"]) =>
+    !!favorites.find((fav: ImagePerBreed) => fav.id === id);
+
+  // const [favorite, setFavorite] = useState(() => {});
```

Comment:
Those comments are probably some leftovers. How did you initially tried to approach this hook though?

Reply:
Ooops. Busted! 😅 

Not many differences to what the actual useFavorite hook is working like. I just started creating this in parallel inside both the CatsModal component and the Favorites page and in the end I moved all the logic to one place to re-use it.

The major difference was that the local storage was part of the implementation and it then moved to a different hook